File: go/solution/0007.go
//
// Solution for Project Euler problem 7
// Copyright michimani All rights reserved.
//
// https://projecteuler.net/problem=7
//
package solution

import "fmt"

// Solve0007 is function to solve the problem.
func Solve0007() {
	var answer int
	var primes []int = []int{}
	var isPrime bool = true
	var num int = 2

	for len(primes) < 10002 {
		isPrime = true

		for _, prime := range primes {
			if num%prime == 0 {
				isPrime = false
				break
			}

		}

		if isPrime {
			primes = append(primes, num)
		}

		num++
	}

	answer = primes[10000]
	fmt.Println(answer)
}

File: python/0007.py
#
# Solution for Project Euler problem 7
# Copyright michimani All rights reserved.
#
# https://projecteuler.net/problem=7
#

TARGET_PRIME_IDX = 10001


def compute() -> int:
    prime_list: list[int] = []
    prime_cnt = 0
    num = 2
    ans = 2
    while prime_cnt < TARGET_PRIME_IDX:
        is_prime = True
        for prime in prime_list:
            if num % prime == 0:
                is_prime = False
                break
        if is_prime is True:
            prime_list.append(num)
            prime_cnt = prime_cnt + 1
            ans = num

        num = num + 1

    return ans


if __name__ == "__main__":
    print(compute())
